src/components/Explorer/queryTemplate.js

```diff
@@ -71,75 +100,90 @@ ${isTi7Team ? 'AND teams.team_id IN (5, 15, 39, 46, 2163, 350190, 1375614, 18383
 GROUP BY hero_id
 ORDER BY total ${(order && order.value) || 'DESC'}`;
   } else {
-    const selectVal = (select && select.groupValue) || (select && select.value) || 1;
-    const groupVal = group ? `${group.value}${group.bucket ? ` / ${group.bucket} * ${group.bucket}` : ''}` : null;
+    const selectVal = {};
+    const groupVal = {};
+    console.log(grouparray);
+    if (groupexists) {
+      grouparray.forEach((x) => { groupVal[x.key] = `${x.value}${x.bucket ? ` / ${x.bucket} * ${x.bucket}` : ''}`; });
+    }
+    if (select) {
+      select.forEach((x) => { selectVal[x.key] = x.groupValue || (x && x.value) || 1; });
+    }
     query = `SELECT
-${select && select.distinct && !group ? `DISTINCT ON (${select.value})` : ''}
-${(group) ?
-    [`${group.groupKeySelect || groupVal} ${group.alias || ''}`,
-      (select && select.countValue) || '',
-      (select && select.avgPerMatch) ? `sum(${selectVal})::numeric/count(distinct matches.match_id) avg` : `${select && select.avg ? select.avg : `avg(${selectVal})`} avg`,
+    ${select ? select.map(x => (x.distinct && grouparray.length < 1 ? `DISTINCT ON (${x.value})` : '')).join('') : ''}
+    ${(groupexists) ?
+    grouparray.map(x =>
+      [`${x.groupKeySelect || groupVal[x.key]} ${x.alias || ''},`].filter(Boolean).join(',\n')).join('') : ''}
+${(groupexists) ? // eslint-disable-line no-nested-ternary
```

Comment:
I recommend not disabling no-nested-ternary, move the code out into another function if needed.